db/commitlog/commitlog.cc

```diff
@@ -1206,7 +1206,7 @@ db::commitlog::segment_manager::segment_manager(config c)
     }())
     , max_size(std::min<size_t>(std::numeric_limits<position_type>::max() / (1024 * 1024), std::max<size_t>(cfg.commitlog_segment_size_in_mb, 1)) * 1024 * 1024)
     , max_mutation_size(max_size >> 1)
-    , max_disk_size(size_t(std::ceil(cfg.commitlog_total_space_in_mb / double(smp::count))) * 1024 * 1024)
+    , max_disk_size(std::max(2 * max_size, size_t(std::ceil(cfg.commitlog_total_space_in_mb / double(smp::count))) * 1024 * 1024))
```

Comment:
How is this related to the added unit test?

Reply:
It is not. See second paragraph of header/title.

Reply:
I think that rather than silently trimming the value to (twice) the max, it'd be better to throw an exception with the details and refuse to start.

Reply:
That would break/change behaviour. I think maybe adding a warning about the threshold being upped in "from_config" (the way scylla options become cl options) is better.

Reply:
Warning added.